File: Mobile-test/app/src/main/java/com/derek/quickseriestest/mobile_test/WebviewActivity.java
package com.derek.quickseriestest.mobile_test;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.webkit.WebView;

/**
 * Created by ddibblee on 8/15/2017.
 */

public class WebviewActivity extends AppCompatActivity {

    WebView webview;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_display_webpage);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        webview = (WebView) findViewById(R.id.webview);

        Intent intent = getIntent();
        String url = intent.getStringExtra("url");
        webview.loadUrl(url);
    }
}

File: app/src/main/java/com/lidboud/kotlin/ui/webview/WebViewActivity.kt
package com.lidboud.kotlin.ui.webview

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import com.lidboud.kotlin.R
import kotlinx.android.synthetic.main.web_layout.custom_webView

class WebViewActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.web_layout)
        val url = intent.getStringExtra(EXTRA_URL)
        custom_webView.webViewClient = getWebClient(false, url)
        custom_webView.settings.useWideViewPort = true
        custom_webView.settings.loadWithOverviewMode = true
        custom_webView.loadUrl(url)
    }

    private fun getWebClient(shouldNavigate: Boolean, url: String) = object : WebViewClient() {
        override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
            return !shouldNavigate || request?.url.toString() == url
        }
    }

    companion object {
        const val EXTRA_URL = "extra_url"
        fun getIntent(context: Context, url: String): Intent {
            val intent = Intent(context, WebViewActivity::class.java)
            intent.putExtra(EXTRA_URL, url)
            return intent
        }
    }
}


